fenzo-core/src/main/java/com/netflix/fenzo/queues/tiered/Tier.java

```diff
@@ -68,10 +73,21 @@ public void queueTask(QueuableTask t) throws TaskQueueException {
 
     @Override
     public QueuableTask nextTaskToLaunch() throws TaskQueueException {
-        for (QueueBucket bucket: sortedBuckets.getSortedList()) {
+        ResAllocs remaining = null;
+        for (QueueBucket bucket : sortedBuckets.getSortedList()) {
             final QueuableTask task = bucket.nextTaskToLaunch();
             if (task != null) {
-                return task;
+                if (bucket.isBelowGuaranteedConsumption()) {
+                    return task;
+                }
+                if (remaining == null) {
+                    ResAllocs totalTierResources = availableTierResGetter.get();
+                    ResAllocs used = ResAllocsUtil.addAll(sortedBuckets.getSortedList().stream().map(QueueBucket::getUsedOrGuaranteed).collect(Collectors.toList()));
```

Comment:
This is an expensive computation to make for every task that is to be assigned resources. The used resources are already kept as a running count in Tier.totals. However, you would need to get the ceiling, which implies that the Tier object would need to maintain two running counts. One is the actual usage, like right now, and another counts the ceiling.